test/hierarchical/test_ordinal.py

```diff
@@ -0,0 +1,310 @@
+from pathlib import Path
+from typing import Dict, List
+
+import numpy as np
+import petab
+import pytest
+
+import pypesto
+import pypesto.logging
+import pypesto.optimize
+import pypesto.petab
+from pypesto.C import LIN, MAX, MODE_FUN, REDUCED, STANDARD, InnerParameterType
+from pypesto.hierarchical.optimal_scaling import (
+    OptimalScalingInnerSolver,
+    OptimalScalingProblem,
+)
+from pypesto.hierarchical.optimal_scaling.parameter import (
+    OptimalScalingParameter,
+)
+from pypesto.hierarchical.optimal_scaling.solver import (
+    compute_interval_constraints,
+    get_surrogate_all,
+)
+
+inner_solver_options = [
+    [
+        {
+            key: value
+            for key, value in zip(
+                ['method', 'reparameterized', 'intervalConstraints'],
+                [method, reparameterized, intervalConstraints],
+            )
+        }
+        for method, reparameterized in zip(
+            ['standard', 'reduced', 'reduced'], [False, False, True]
+        )
+    ]
+    for intervalConstraints in ['max', 'max-min']
+]
+
+example_ordinal_yaml = (
+    Path(__file__).parent
+    / '..'
+    / '..'
+    / 'doc'
+    / 'example'
+    / 'example_ordinal'
+    / 'example_ordinal.yaml'
+)
+
+
+@pytest.fixture(params=inner_solver_options)
+def inner_solver_options(request):
+    return request.param
+
+
+def test_evaluate_objective(inner_solver_options: List[Dict]):
+    """Check that standard / reduced / reparameterized formulations yield the
+    same result."""
+    petab_problem = petab.Problem.from_yaml(example_ordinal_yaml)
+    vals = []
+    for idx, option in enumerate(inner_solver_options):
+        problem = _create_problem(petab_problem, option)
+        val = problem.objective(np.array([0, 0]))
+        vals.append(val)
+        assert np.isclose(vals[idx], vals[idx - 1])
+
+
+def test_optimization(inner_solver_options: List[Dict]):
+    """Check that optimizations finishes without error."""
```

Comment:
you could add a check whether 
* the obtained values (fval, x, grad, ...) are not nan or inf (`np.isfinite, np.allfinite`)
* whether the final fval is < fval0